### PR TITLE
tests: use ceph-ansible to deploy, configure and verify test builds

### PR DESCRIPTION
**Pull request by andrewschoen**
Let's not merge this until we can change the CI to run the tests against this PR. I should be able to get that done today. I have tested this manually though and it does work on one of our jenkins slaves.

The idea is that for each PR we will start a CI job that builds the images, starts a local insecure docker registry and then adds the new images to that registry. The script then uses ceph-ansible and it's existing ``docker_cluster`` testing scenario to deploy, configure and test a ceph cluster using the new image from the registry on the jenkins slave.

### CODE REVIEW

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

File: tests/setup.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+
+- hosts: all
+  gather_facts: true
+  tasks :
+    - name: allow nodes to use an insecure registry
+      sudo: yes
```
Comment (leseb) [inline tests/setup.yml]: `s/sudo/become/`

File: tests/setup.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      blockinfile:
+        dest: /etc/sysconfig/docker
+        block: |
+          INSECURE_REGISTRY='--insecure-registry {{ public_network }}'
+        marker: "# {mark} ANSIBLE MANAGED BLOCK" 
+
+    - name: restart docker
+      sudo: yes
```
Comment (leseb) [inline tests/setup.yml]: ditto

File: travis-builds/build_imgs.sh [hunk 3]
```diff
@@ -34,23 +34,23 @@ function copy_dirs {
 
 function build_base_img {
   pushd base
-  docker build -t base .
+  docker build -t ceph/base .
```
Comment (leseb) [inline travis-builds/build_imgs.sh]: This obviously breaks Travis but I guess we don't care as this should replace Travis in the long term.

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

**Comment by andrewschoen** [comment]
I've got this passing now on the slaves in our sepia labs, but the nodes we use in OVH are not passing the tests. I'm investigating why.

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

**Comment by alfredodeza** [comment]
jenkins test this please

**Comment by andrewschoen** [comment]
jenkins test ceph_ansible2.1-jewel-centos7_cluster

**Comment by andrewschoen** [comment]
Alright, this seems to be stable now. This does need https://github.com/ceph/ceph-ansible/pull/1229 to be able to use a custom registry though. We can leave this commit https://github.com/ceph/ceph-docker/pull/458/commits/189d4383c23a7662daaaca2e259bdca7631b5bad in here for now, but eventually once that's merged we'll want to backport it to the stable-2.1 branch so we can go back to using it for testing.

**Pull request merged**